### PR TITLE
[MIG] contract_membership_delegated_partner: Migration to 11.0

### PR DESCRIPTION
**Pull request by chienandalu**
This module allows to set the delegated membership at the contract so the
recurring invoices automatically have it set.

cc @Tecnativa

### CODE REVIEW

**Comment by chienandalu** [comment]
@rafaelbn @pedrobaeza Can you review here, please?

**Review by pedrobaeza — APPROVED** [review]

**Review by yajo — APPROVED** [review]

File: contract_membership_delegated_partner/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+* David Vidal <david.vidal@tecnativa.com>
```
Comment (rafaelbn) [inline contract_membership_delegated_partner/readme/CONTRIBUTORS.rst:1]: IMHO I have dedicated hours enough to be here!

File: contract_membership_delegated_partner/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#. Set the delegated partner in contract.
```
Comment (rafaelbn) [inline contract_membership_delegated_partner/readme/USAGE.rst:1]: Please add that you should go to user and enable "Delegated partner in membership" group for using this functionality.

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Hi @chienandalu 

- Go to runbot, check this contract and try to create the invoice

http://3328470-61-9b6374.runbot2.odoo-community.org/web?debug#id=18&view_type=form&model=account.analytic.account&menu_id=145&action=229

- You will get this error

```
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 935, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 923, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/dependencies/contract/contract/models/account_analytic_account.py", line 298, in recurring_create_invoice
    invoices |= contract.with_context(ctx)._create_invoice()
  File "/home/odoo/dependencies/contract/contract/models/account_analytic_account.py", line 266, in _create_invoice
    self.env['account.invoice.line'].create(invoice_line_vals)
  File "/home/odoo/build/OCA/vertical-association/membership_variable_period/models/account_invoice.py", line 67, in create
    line = super(AccountInvoiceLine, self).create(vals)
  File "/home/odoo/build/OCA/vertical-association/membership_prorate/models/account_invoice.py", line 53, in create
    invoice_line)
  File "/home/odoo/build/OCA/vertical-association/membership_prorate/models/account_invoice.py", line 30, in _prepare_invoice_line_prorate_vals
    if date_invoice < date_from:
TypeError: unorderable types: datetime.date() < NoneType()
```

File: contract_membership_delegated_partner/readme/USAGE.rst (hunk 2 above)
Comment (chienandalu) [reply contract_membership_delegated_partner/readme/USAGE.rst:1]: It's in the CONFIGURE.rst section

**Comment by chienandalu** [comment]
@rafaelbn Rebasing after merging of `membership_prorate_variable_period` wich absence is probably the cause of this error

**Comment by yajo** [comment]
@rafaelbn update your review please.

**Review by rafaelbn — APPROVED** [review]
:+1:

**Pull request merged**